### PR TITLE
chore: fix clippy warn

### PR DESCRIPTION
**Pull request by driftluo**
Within CI, using the latest `fmt` and `clippy`, but allow it fail

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -115,13 +117,17 @@ docker-publish:
 ##@ Code Quality
 .PHONY: fmt
 fmt: setup-ckb-test ## Check Rust source code format to keep to the same style.
+	echo ${CLIPPY_FMT_VER} > rust-toolchain
```
Comment (yangby-cryptape) [inline Makefile:120]: - Remove all `echo` statements.

- Use a input parameter:

  ```Makefile
  cargo +${CHECK_VERSION} fmt ${VERBOSE} --all -- --check
  cargo +${CHECK_VERSION} clippy ${VERBOSE} --all -- --check
  ```

File: .travis.yml [hunk 2]
```diff
@@ -71,6 +71,9 @@ matrix:
       if: 'tag IS NOT present AND (type = pull_request OR branch in (master, staging, staging2, trying) OR repo != nervosnetwork/ckb)'
       os: linux
       install:
+        - rustup update stable
```
Comment (doitian) [inline .travis.yml:74]: Please setup a new job to check fmt/clippy with the latest toolchain and allow failures in this job.

File: .travis.yml [hunk 3]
```diff
@@ -77,6 +79,21 @@ matrix:
...
+    - name: Latest Linters
+      if: 'tag IS NOT present AND (type = pull_request OR branch in (master, staging, staging2, trying) OR repo != nervosnetwork/ckb)'
+      os: linux
+      rust: stable
+      cache: false
+      addons: { apt: { packages: [] } }
+      env:
+        - CLIPPY_OPTS="-D warnings -D clippy::clone_on_ref_ptr -D clippy::enum_glob_use -D clippy::fallible_impl_from"
+      install:
+        - rustup update stable
```
Comment (yangby-cryptape) [inline .travis.yml:91]: Unnecessary since `rust: stable`.

**Comment by doitian** [comment]
Is the new added CI job supposed to pass in this PR? Now it failed.

**Comment by driftluo** [comment]
> Is the new added CI job supposed to pass in this PR? Now it failed.

This is intentional. This failure is due to a newly added lint rule, which is not available in the old version. What you need to pay attention to when upgrading toolchain

**Comment by doitian** [comment]
bors r=doitian,yangby-cryptape

**Pull request merged**